### PR TITLE
Photometric series plotting and more

### PR DESCRIPTION
**Pull request by guynir42**
- add example photometric series from TESS
- add a doc file about uploading and downloading photometric series
- fix some issues in parsing the photometric series
- plot the photometric series alongside the regular photometry
- add a TESS filter using an upgrade of sncosmo
- add a TESS instrument to the demo data
- add the photometric series in the demo data

### CODE REVIEW

File: tools/data_loader.py
Comment (mcoughlin) [inline tools/data_loader.py:209]: Should this also have its own try: except?

**Review by mcoughlin — COMMENTED** [review]
Just reading the code a bit. Hopefully the TESS stuff will happen in the next week or so.

File: tools/data_loader.py
Comment (guynir42) [reply tools/data_loader.py:209]: Not really. All the exceptions are caught at the end and then there's an error log. 
I put an exception clause here just to be able to fish out the specific h5 files that I was making for the photometric series (although anyone using the `pd.HDFStore` method will be able to use this). 
If anything goes wrong reading that file it would just put an `X` there and report the error at the end.

Comment (mcoughlin) [reply tools/data_loader.py:209]: Ok!

**Comment by guynir42** [comment]
I should mention that this is still a work in progress.
The code I have now seems to work fine, but I have a bunch more to add...

**Comment by mcoughlin** [comment]
@guynir42 I should say I was just reading with interest, I didn't mean it as a review, just what I noticed.

**Comment by guynir42** [comment]
You can even review what is here, but just be aware that more is coming...

**Review by profjsb — APPROVED** [review]
Looks good. You probably need to add photometric_series to the `index.rst` in `doc/` for it to show up? Also can you add a screenshot to this PR? Will make it easier to see what this PR is actually enabling on the front end.

File: doc/photometric_series.md
Comment (mcoughlin) [inline doc/photometric_series.md:19]: Also can pass in mag and magerr?

File: doc/photometric_series.md
Comment (mcoughlin) [inline doc/photometric_series.md:169]: Will this be continued?

File: skyportal/handlers/api/photometry.py
Comment (mcoughlin) [inline skyportal/handlers/api/photometry.py:1383]: Any sorting needed here? Could be nice to sort by date?

File: skyportal/handlers/api/source.py
Comment (mcoughlin) [inline skyportal/handlers/api/source.py:1410]: Needed?

File: skyportal/tests/api/test_photometric_series.py
Comment (mcoughlin) [inline skyportal/tests/api/test_photometric_series.py:2138]: What is this for exactly?

**Review by mcoughlin — APPROVED** [review]
A few small things.

File: skyportal/tests/api/test_photometric_series.py
Comment (guynir42) [reply skyportal/tests/api/test_photometric_series.py:2138]: Ahh yes, I forgot quite a few things in there while I was working... Good catches on all of these. 

I am thinking about leaving this in until we get the TESS filter merged so I can safely add the demo data with the correct filter...

Comment (guynir42) [reply skyportal/tests/api/test_photometric_series.py:2138]: Let me make this even more obvious so it doesn't get left in there by mistake.

**Review by profjsb — CHANGES_REQUESTED** [review]
I was able to render the plot in your PR. However a few things stuck out. 

- First, the filter name on the plot seems weird (TESS/kepler). I wasn't aware that TESS used Kepler filters. Perhaps fix or clarify.
- Second, given all the data points, I wanted to change the default binning but click on the binsize slider the browser hangs.
- Third, I'm worried about the unbounded number of datapoints that could be plotted if the timeseries is very large. I asked chatgpt for help here. Here's what I got back: 

**I have more than 5000 data points to plot using bokeh. What's the best way to do this so that I do not slow down my browser?**

<i>When working with a large amount of data points, it's important to consider the performance of your visualization. Bokeh provides several techniques to optimize your visualization and avoid slowing down your browser.

Here are a few tips to get started:

Use downsampling techniques to reduce the number of data points displayed on the plot. Bokeh provides several built-in downsampling methods, such as "auto" and "subsample", which can be used to reduce the number of points displayed based on various criteria. For example, the "subsample" method will randomly select a subset of the data points to display, while the "auto" method will automatically select a downsample method based on the current plot size and the number of data points.

Use glyphs that are optimized for large datasets. Bokeh provides several glyphs that are optimized for displaying large datasets, such as "line" and "circle". These glyphs are designed to be efficient in terms of rendering performance, and can handle a large number of data points without slowing down the browser.

Use webgl backend. By default Bokeh uses canvas backend, but you can enable webgl backend by setting output_backend='webgl' on your output_file or output_notebook. Webgl is faster in general, but can also handle larger datasets, specially when using tools like Scatter or HeatMap.

Use server-side rendering. If downsampling and webgl are not enough and you still need to display a large dataset, you can use Bokeh's server-side rendering capabilities to generate the plot on the server and display it in the browser as a static image or interactive BokehJS document.

By using these techniques, you can create high-performance visualizations that can handle even very large datasets without slowing down your browser.</i>

**Comment by guynir42** [comment]
@profjsb The filter is weird because I need to pin the sncosmo version that has TESS. For now Kepler is just a placeholder. 

Regarding the number of data points, I will give a try to the `auto` feature and ping you to have another look.

**Comment by profjsb** [comment]
copilot:all

**Comment by profjsb** [comment]
copilot:all

**Comment by profjsb** [comment]
@guynir42 can you resolve the conflict so we can merge?

**Comment by guynir42** [comment]
@profjsb I will resolve it, but we should not merge until the sncosmo updates with the TESS filter. Then I can put that in the demo data.

**Comment by guynir42** [comment]
@profjsb @mcoughlin test failures appear to be unrelated. Does any of you want to have another look before I merge?

**Pull request merged**
